pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -561,29 +578,21 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 				delete(specKubeletConfig.SystemReserved, "cpu")
 			}
 
+			// FeatureGates must be set from the FeatureGate.
+			// Remove them here to prevent the specKubeletConfig merge overwriting them.
+			specKubeletConfig.FeatureGates = nil
```

Comment:
where does the FeatureGates get injected in for this object?

Reply:
Line 543 uses a new kubelet generator function as part of the refactor, which is shared between this and the feature gate handler, which handles the feature gate injection
```
originalKubeConfig, err := generateOriginalKubeletConfigWithFeatureGates(cc, ctrl.templatesDir, role, features)
```
If you look at https://github.com/openshift/machine-config-operator/pull/2668/commits/1b4429b675618d4fdfacb8c9aa19ebd77cde53c8 explicitly it's a bit easier to see what the change here is

I could expand this comment to make this more clear
```
// FeatureGates must be set from the FeatureGate.
// Remove them here to prevent the specKubeletConfig merge overwriting them.
// The originalKubeConfig already has these merged in from the FeatureGate.
```